lib/Fmt_ast.ml

```diff
@@ -557,6 +564,26 @@ let rec fmt_extension c ctx key (ext, pld) =
   | _, _, PSig [({psig_desc= Psig_type _; _} as si)], (Pld _ | Sig _ | Top)
     ->
       fmt_signature_item c ~ext (sub_sig ~ctx si)
+  (* Quoted extensions (since ocaml 4.11) {%ext delim|...|delim} *)
+  | ( ("%" | "%%")
+    , ext
+    , PStr
+        [ { pstr_desc=
+              Pstr_eval
+                ( { pexp_desc= Pexp_constant (Pconst_string (str, _, delim))
+                  ; pexp_loc
+                  ; pexp_loc_stack= _
+                  ; pexp_attributes }
+                , attrs )
+          ; pstr_loc } ]
+    , _ )
+    when Source.is_quoted_string c.source pstr_loc ->
+      let doc, atrs = doc_atrs attrs in
+      fmt_docstring c doc
+      $ Cmts.fmt c pstr_loc @@ hvbox 0 @@ Cmts.fmt c pexp_loc @@ hvbox 0
```

Comment:
None of the tests are triggering this code but are instead testing attributes and comments around expressions and structure items.
I suggest to remove the tests with attributes and comments (or keep a single one) to avoid confusion.

Looking at the parser I couldn't find any input that would add attributes or comments here. This may be added in the future but this code would most probably move them (which is a fatal error) because it clashes with the syntax for comments/attributes around expressions.
I suggest to remove this code and to add an assert that there is no comments (necessary to avoid the warning "Some locations have not been considered").

Reply:
That's a good point, I added the assertions. (please nobody suggest the compiler team to allow this in the parser)
I cleared up some lines in the tests, but I like to keep a good number with attributes and comments, to have them updated in case we change something else (box, attrs, comment attachment algorithm)

Reply:
More tests don't harm.